rcl_yaml_param_parser/src/parser.c

```diff
@@ -35,40 +35,103 @@
 #include "./impl/node_params.h"
 #include "./impl/yaml_variant.h"
 
-#define MAX_NUM_NODE_ENTRIES 256U
+#define INIT_NUM_NODE_ENTRIES 256U
 
 ///
 /// Create the rcl_params_t parameter structure
 ///
 rcl_params_t * rcl_yaml_node_struct_init(
   const rcutils_allocator_t allocator)
+{
+  return rcl_yaml_node_struct_init_with_capacity(allocator, INIT_NUM_NODE_ENTRIES);
+}
+
+rcl_params_t * rcl_yaml_node_struct_init_with_capacity(
+  const rcutils_allocator_t allocator,
+  size_t capacity)
 {
   RCUTILS_CHECK_ALLOCATOR_WITH_MSG(&allocator, "invalid allocator", return NULL);
+  if (capacity == 0) {
+    RCUTILS_SET_ERROR_MSG("capacity can't be zero");
+    return NULL;
+  }
   rcl_params_t * params_st = allocator.zero_allocate(1, sizeof(rcl_params_t), allocator.state);
   if (NULL == params_st) {
-    RCUTILS_SAFE_FWRITE_TO_STDERR("Error allocating mem\n");
-    return NULL;
+    RCUTILS_SET_ERROR_MSG("Error allocating mem");
+    goto clean;
```

Comment:
```suggestion
    return NULL;
```

(we haven't allocated anything at this point, so this is safe to short-circuit here)